core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -1444,12 +1433,12 @@ private void searchForTitle(String title, boolean openInNewTab) {
 
   @Override
   public void onActivityResult(int requestCode, int resultCode, Intent data) {
-    boolean openSearchInANewTab = isInTabSwitcher(); // check if in tab mode
-    hideTabSwitcher();
     Log.i(TAG_KIWIX, "Intent data: " + data);
     switch (requestCode) {
       case MainMenuKt.REQUEST_FILE_SEARCH:
         if (resultCode == RESULT_OK) {
+          boolean isFromTabSwitcher = mainMenu.isInTabSwitcher();
```

Comment:
this doesn't need to be declared here, inline it

Reply:
@macgills the call to `hideTabSwitcher();` in the next line will clear this boolean value so I need to save it before entering the if condition

Reply:
then the boolean should be called `wasFromTabSwitcher` maybe?